edgelet/iotedge/src/support_bundle.rs

```diff
@@ -0,0 +1,230 @@
+        SupportBundle {
+            runtime,
+            log_options,
+            location: location.to_owned(),
+        }
+    }
+
+    fn make_state(self) -> Result<BundleState<M>, Error> {
+        /* Print status */
+        let since_time: DateTime<Utc> = DateTime::from_utc(
+            NaiveDateTime::from_timestamp(self.log_options.since().into(), 0),
+            Utc,
+        );
+        let since_local: DateTime<Local> = DateTime::from(since_time);
+        let max_lines = if let LogTail::Num(tail) = self.log_options.tail() {
+            format!("(maximum {} lines) ", tail)
+        } else {
+            "".to_owned()
+        };
+        println!(
+            "Writing all logs {}since {} (local time {}) to {}",
+            max_lines, since_time, since_local, self.location
+        );
+
+        /* Make state */
+        let file_options =
+            zip::write::FileOptions::default().compression_method(zip::CompressionMethod::Deflated);
+
+        let zip_writer = zip::ZipWriter::new(
+            File::create(format!("{}/css_bundle.zip", self.location.to_owned()))
+                .map_err(|err| Error::from(err.context(ErrorKind::WriteToStdout)))?,
+        );
+
+        Ok(BundleState {
+            runtime: self.runtime,
+            log_options: self.log_options,
+            location: self.location,
+            file_options,
+            zip_writer,
+        })
+    }
+
+    fn write_all_logs(s1: BundleState<M>) -> impl Future<Item = BundleState<M>, Error = Error> {
+        SupportBundle::get_modules(s1).and_then(|(names, s2)| {
+            stream::iter_ok(names).fold(s2, SupportBundle::write_log_to_file)
+        })
+    }
+
+    fn get_modules(
+        state: BundleState<M>,
+    ) -> impl Future<Item = (Vec<String>, BundleState<M>), Error = Error> {
+        state
+            .runtime
+            .list_with_details()
+            .map_err(|err| Error::from(err.context(ErrorKind::ModuleRuntime)))
+            .map(|(module, _s)| module.name().to_string())
+            .collect()
+            .map(|names| (names, state))
+    }
+
+    fn write_log_to_file(
+        state: BundleState<M>,
+        module_name: String,
+    ) -> impl Future<Item = BundleState<M>, Error = Error> {
+        println!("Writing {} to file", module_name);
+        let BundleState {
+            runtime,
+            log_options,
+            location,
+            file_options,
+            mut zip_writer,
+        } = state;
+
+        let file_name = format!("logs/{}_log.txt", module_name);
+        zip_writer
+            .start_file_from_path(&Path::new(&file_name), file_options)
+            .into_future()
+            .map_err(|err| Error::from(err.context(ErrorKind::WriteToStdout)))
+            .and_then(move |_| {
+                pull_logs(&runtime, &module_name, &log_options, zip_writer).map(move |zw| {
+                    println!("Wrote {} to file", module_name);
```

Comment:
We are probably going to need to do a quick "strings" review on this for all of the text written to stdout. We'll have to loop in whoever is PM'ing this.